### PR TITLE
Status update

### PR DESCRIPTION
**Pull request by Benjamin-Lee**
Adds information about the current status of the project (we are no longer actively soliciting new rules as much as combining the rules we already have) to the README as well as clarifies when to PR vs commit.

Can someone with experience from the deep review make sure that the contributing policy is clear/useful?

### CODE REVIEW

**Comment by Benjamin-Lee** [comment]
The PR clarification is in response to @rasbt's question in #54, by the way.

File: README.md [hunk 1]
```diff
@@ -27,36 +27,41 @@ Join us!
...
+- Help organize the [proposed rules](https://github.com/Benjamin-Lee/deep-rules/issues?q=label%3Arule) into one [document](https://github.com/Benjamin-Lee/deep-rules/blob/master/rules.md).
+- Know of anyone who might be interested? [Send them an email!](mailto:@?subject=Ten%20Simple%20Rules%20for%20Using%20Deep%20Learning%20in%20Biology%20paper&body=Take%20a%20look%20at%20the%20Deep%20Rules%20project%2C%20a%20PLOS%20Computational%20Biology%20paper%20written%20collaboratively%20using%20GitHub!%20%0A%0Ahttps%3A%2F%2Fgithub.com%2FBenjamin-Lee%2Fdeep-rules%0A%0AIf%20you%20know%20of%20anyone%20who%20might%20be%20interested%2C%20feel%20free%20to%20forward%20this%20email%20along!)
 - Have Twitter followers? [Tweet about the project!](https://twitter.com/intent/tweet?text=Have%20you%20used%20deep%20learning%20in%20your%20research%3F%20If%20so%2C%20contribute%20to%20the%20Ten%20Simple%20Rules%20for%20Deep%20Learning%20in%20Biology%20paper%21%0A%0Ahttp%3A%2F%2Fgithub.com%2FBenjamin-Lee%2Fdeep-rules)
```
Comment (dhimmel) [inline README.md]: Can we make the github link in the proposed tweet HTTPS? BTW that is cool you can do that.

File: README.md [hunk 2]
```diff
@@ -27,36 +27,41 @@ Join us!
...
 
 ## Manubot
 
 <!-- usage note: do not edit this section -->
 
-Manubot is a system for writing scholarly manuscripts via GitHub.
-Manubot automates citations and references, versions manuscripts using git, and enables collaborative writing via GitHub.
-The [Manubot Rootstock repository](https://git.io/vQSvo) is a general purpose template for creating new Manubot instances, as detailed in [`SETUP.md`](SETUP.md).
-See [`USAGE.md`](USAGE.md) for documentation how to write a manuscript.
+Manubot is a system for writing scholarly manuscripts via GitHub. Manubot
```
Comment (dhimmel) [inline README.md]: Suggestion not to edit this section to avoid future conflicts when updating manubot.

Looks like the edits are mostly changing one line per sentence to word wrap... one sentence per line is highly recommended.

**Comment by evancofer** [comment]
The contribution policy makes sense. I agree with @agitter that detailing the PR process in `CONTRIBUTING.md`. Using PRs for everything also seems like a pretty good way to scale up.

File: README.md (hunk 2 above)
Comment (Benjamin-Lee) [reply README.md]: Didn't know this could be an issue; reverted.

**Comment by Benjamin-Lee** [comment]
@dhimmel, one line per sentence is done!

**Comment by Benjamin-Lee** [comment]
@agitter and @evancofer, PRs for everything works for me. I removed the option for contributors to bypass PRs in CONTRIBUTING.md as well as added the overview of the PR protocol.

What is the advantage of using `[MRG+1]` after approval versus using GH's approval feature? Not against it, just don't fully understand it.

**Comment by rasbt** [comment]
> What is the advantage of using [MRG+1] after approval versus using GH's approval feature? Not against it, just don't fully understand it.

I see these two as useful but slightly different. The approval feature is more like to provide direct feedback on the suggested commit. (There may be many rounds of commits and re-approval).
The [MRG] prefix or suffix in the title indicates that something is "ready" to be merged (whereas [WIP] indicates that this is currently still being worked on). It's actually very useful to use these tags in the title because it makes it easier to identify these things when going to the issue tracker. 

So while [MRG] indicates that something is ready to be merged, + 1 then indicates that another person approved it (to ensure we have at least a second opinion on this), and by someone else then doing the actual merging of [MRG] +1 tagged PRs, we hopefully make extra sure that everything is in order before merging.


> - New pull request (optionally with [WIP]?)
> - Pull request author adds [MRG] to the pull request title to indicate it is ready to merge and for review
> - Reviewer edits [MRG] to [MRG+1] after approval
> - Maintainer merges

That sounds good!

> If this scales up further, I also recommend making pull requests for everything. Even if you want to self-approve and merge trivial changes, it broadcasts typo fixes or other minor changes to everyone watching the repository.

Yes, I think this would be a good practice given that we have so many contributors.

**Comment by agitter** [comment]
> What is the advantage of using [MRG+1] after approval versus using GH's approval feature?

I believe the value is for massive projects like scikit-learin that have 100s of open [pull requests](https://github.com/scikit-learn/scikit-learn/pulls).  Maybe they use it to quickly scroll through pull requests and identify those ready for review or merge?  Or to provide standard strings for searching?  @rasbt may know better.

This project will have fewer concurrent pull requests, so setting the policy of approval in the pull request and then a separate maintainer for the merge could suffice.

**Comment by rasbt** [comment]
> I believe the value is for massive projects like scikit-learin that have 100s of open pull requests. Maybe they use it to quickly scroll through pull requests and identify those ready for review or merge? Or to provide standard strings for searching? @rasbt may know better.

Yes that's true regarding big projects. However, I see the approval feature more as a "per-commit" review feature whereas the MRG refers to the PR as a whole. 

Or wait... are we talking about an automatic feature implemented in the GitHub UI that automatically requires Reviewer approval like in the PyTorch repo? 



Just a screenshot from a random issue there (https://github.com/pytorch/pytorch/pull/14278): 

screen shot 2018-11-21 at 12 08 48 pm

I thought this was a plugin for some reason. If we can enable a "review required" by default upon opening a PR (like in the PyTorch repo) I think this would be even better than the [MRG] tag (which is more oldschool)

**Comment by evancofer** [comment]
@rasbt We can indeed automatically require approval before a PR is merged (described in detail here: https://help.github.com/articles/enabling-required-reviews-for-pull-requests/ ).

**Comment by agitter** [comment]
I wasn't aware of GitHub's required reviews.  Nice feature.

Using required reviews sounds good to me.  If we want to use that instead of `[MRG]` annotations, we should update that section of `The Contribution Process` before merging.

**Comment by Benjamin-Lee** [comment]
Before I activate the feature, here's the settings that (I think) represent what we want:

image

Having two required approving reviews seems to be equivalent to a second reviewer merging the PR with `[MRG]+1`; instead they mark their approval and then merge.

**Comment by rasbt** [comment]
Looks good! 

> Having two required approving reviews seems to be equivalent to a second reviewer merging the PR with [MRG]+1; instead they mark their approval and then merge.

Yes, I think that's correct.

**Comment by agitter** [comment]
> Having two required approving reviews seems to be equivalent to a second reviewer merging the PR with [MRG]+1; instead they mark their approval and then merge.

Yes, that makes sense to me.  The second reviewer can merge as soon as they indicate their approval.

Will we loose the ability to self-review and merge typo fix pull requests under this configuration?  I'm not concerned about that if it does happen.  With this many maintainers, small pull requests will be reviewed quickly.

**Comment by agitter** [comment]
@Benjamin-Lee before merging this, let's update the contribution process guide.  Do we want something like:

> 1. Contributor creates a new pull request (optionally with `[WIP]` in the title to indicate work in progress)
> 2. Pull request author removes [`WIP`] to indicate it is ready for review
> 3. Maintainer suggests edits and approves the pull request after all changes are made
> 4. Second maintainer approves and merges the pull request

**Comment by Benjamin-Lee** [comment]
Ok, I updated `CONTRIBUTING.md`. 

As an aside, to members of the team in the US, happy Thanksgiving! I'll be AFK for the rest of the day and I hope you all get the opportunity to have a nice day off as well.

**Comment by Benjamin-Lee** [comment]
Branch protection for `master` is active –  this PR should now require two approvals before merging.

**Comment by agitter** [comment]
I approved this pull request.  Anyone else want to approve and merge?
